lib/mix/tasks/openapi.json.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Mix.Tasks.Openapi.Spec.Json do
+  @moduledoc """
+  Serialize the given OpenApi spec module to a json file.
+  """
+  use Mix.Task
+  require Mix.Generator
+
+  @default_filename "openapi.json"
+
+  defmodule Options do
+    @moduledoc false
+
+    defstruct filename: nil, spec: nil, endpoint: nil, pretty: false
+  end
+
+  @impl true
+  def run(argv) do
+    opts = parse_options(argv)
+
+    with :ok <- maybe_start_endpoint(opts),
+         {:ok, content} <- generate_spec(opts) do
+      write_spec(content, opts.filename)
+    else
+      {:error, error} -> raise error
+    end
+  end
+
+  def maybe_start_endpoint(%{endpoint: nil}) do
+    :ok
+  end
+
+  def maybe_start_endpoint(options) do
+    if Code.ensure_loaded?(options.endpoint) do
+      case options.endpoint.start_link() do
+        {:ok, _} ->
+          :ok
+
+        _ ->
+          {:error, "Couldn't start endpoint"}
+      end
+    else
+      {:error, "Module #{options.endpoint} is not a valid Phoenix Enpoint"}
+    end
+  end
+
+  def generate_spec(options) do
+    if Code.ensure_loaded?(options.spec) do
+      json_encoder = OpenApiSpex.OpenApi.json_encoder()
+      spec = options.spec.spec()
+      {:ok, spec |> json_encoder.encode!(pretty: options.pretty)}
```

Comment:
Here, and on line 61: Can you eliminate the use of the pipeline operator if it's only used once?

Here is a popular Elixir code style guideline that describes the preference:

https://github.com/christopheradams/elixir_style_guide#avoid-single-pipelines

Reply:
Good catch, I had forgotten it, I know and follow the elixir's guideline, 👍.